src/com/palmergames/bukkit/towny/command/NationCommand.java

```diff
@@ -1192,6 +1197,9 @@ public void nationAdd(Player player, String[] names) throws TownyException {
 		if (TownySettings.getMaxTownsPerNation() > 0 && nation.getTowns().size() >= TownySettings.getMaxTownsPerNation())
 			throw new TownyException(Translatable.of("msg_err_nation_over_town_limit", TownySettings.getMaxTownsPerNation()));
 
+        if (TownySettings.getMaxResidentsPerNation() > 0 && nation.getResidents().size() >= TownySettings.getMaxResidentsPerNation())

```

Comment:
Again this should be checking the future nation.getResidents().size() with the new town added to it.